lib/annealing/metal.rb

```diff
@@ -3,21 +3,21 @@
 module Annealing
   # It manages the total energy of a given collection
   class Metal
+    include Configuration::Configurator
     attr_reader :state, :temperature
 
-    def initialize(state, temperature, energy_calculator: nil, state_change: nil)
-      @state = state
-      @temperature = temperature
-      @energy_calculator = energy_calculator || default_energy_calculator
-      @state_change = state_change || default_state_change
+    def initialize(current_state, current_temperature, config_hash = {})
```

Comment:
Also what do you think of having the config hash like this 
```ruby
    def initialize(current_state, current_temperature, **config)
```

Reply:
I don't really have any strong objection to this idea, but I'm not sure I see the benefit to using that over the positional argument with a default. In both cases they are optional. I suppose the double splat enforces that it must be a hash. I'll make the change as requested, but I'd love to hear your thoughts on why you prefer one over the other if you have a moment.

Reply:
The ** will enable any keyword arguments in a method, while the {} force to receive the hash argument. As my personal preference it depends the case. For this scenario seemed more suitable to receive any keyword argument